packages/rum-core/src/domain/rumSessionManager.ts

```diff
@@ -102,13 +106,15 @@ function hasValidRumSession(trackingType?: string): trackingType is RumTrackingT
   return (
     trackingType === RumTrackingType.NOT_TRACKED ||
     trackingType === RumTrackingType.TRACKED_WITH_SESSION_REPLAY ||
-    trackingType === RumTrackingType.TRACKED_WITHOUT_SESSION_REPLAY
+    trackingType === RumTrackingType.TRACKED_WITHOUT_SESSION_REPLAY ||
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏could we add this new type in the related tests?

It would be nice to ensure somewhere that `sessionReplayAllowed` is false for the forced state as it is used to compute `sampled_for_replay`

Reply:
This is just a typeguard IIRC, `sessionReplayAllowed` compares the tracking type, and is set to true only if the state is `TRACKED_WITH_SESSION_REPLAY`

Reply:
I added a new attribute in the session object that reflect either the session was originally sampled for replay or not, to avoid any confusing (and to fix a small bug I found)

Done ✔️